src/gui/cocoainitializer_mac.mm

```diff
@@ -12,28 +12,51 @@
  * for more details.
  */
 
+#include "application.h"
 #include "cocoainitializer.h"
 
 #import <Foundation/NSAutoreleasePool.h>
 #import <AppKit/NSApplication.h>
 
+#include <QMessageBox>
+
+@interface OwnAppDelegate : NSObject <NSApplicationDelegate>
+- (BOOL)applicationShouldHandleReopen:(NSApplication *)sender hasVisibleWindows:(BOOL)flag;
+@end
+
+@implementation OwnAppDelegate {
+}
+
+- (BOOL)applicationShouldHandleReopen:(NSApplication *)sender hasVisibleWindows:(BOOL)flag
+{
+    if (auto app = dynamic_cast<OCC::Application *>(QApplication::instance()))
+        app->showSettingsDialog();
```

Comment:
Hmm qobject_cast?
Nitpick I prefer {} for all if's

Reply:
done